Update get_product_data to send request with a locale query param for translation

chihsuan opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

This PR updates the `get_product_data` method to get translated contents from WCCOM.

### How to test the changes in this Pull Request:

1. Use a fresh site
2. Go to `Settings > General`
3. Change site language to `Español`
4. Go to OBW
5. Go to Product types step
6. Click on an info icon `(i)`
7. Observe that tooltip shows translated contents.

![Screen Shot 2022-09-01 at 10 50 08](https://user-images.githubusercontent.com/4344253/187822142-ac6107dd-979b-4263-a63d-62fb51685558.png)

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

chihsuan committed:
Update get_product_data to send request with a locale query param for translation

chihsuan committed:
Add changelog

rjchow reviewed (commented):
This looks good! Thanks for working on these i18n details, it makes the product look a lot more polished

BTW - I had to clear the `_transient_wc_onboarding_product_data` option if I changed the language after installing woocommerce. The transient doesn't get invalidated upon language change.

I wonder if we should set the locale as part of the transient and re-fetch it it doesn't match?

chihsuan committed:
Fix lint

chihsuan committed:
Update PRODUCT_DATA_TRANSIENT format for translations

chihsuan commented:
> BTW - I had to clear the `_transient_wc_onboarding_product_data` option if I changed the language after installing woocommerce. The transient doesn't get invalidated upon language change.
> 
> I wonder if we should set the locale as part of the transient and re-fetch it it doesn't match?

Sounds good! @rjchow I've updated the transient format in c123f41740679422026475110b77ec0c3dfd1a3d.

rjchow reviewed (approved):
Fantastic work, thank you!
